metricbeat/module/system/process_summary/process_summary_test.go

```diff
@@ -56,26 +57,18 @@ func TestFetch(t *testing.T) {
 	event, ok := summary.(common.MapStr)
 	require.True(t, ok)
 
-	if runtime.GOOS == "windows" {
-		assert.Contains(t, event, "total")
-		assert.Contains(t, event, "sleeping")
-		assert.Contains(t, event, "running")
-		assert.Contains(t, event, "unknown")
-		total := event["sleeping"].(int) + event["running"].(int) + event["unknown"].(int)
-		assert.Equal(t, event["total"].(int), total)
-	} else {
-		assert.Contains(t, event, "total")
-		assert.Contains(t, event, "sleeping")
-		assert.Contains(t, event, "running")
-		assert.Contains(t, event, "idle")
-		assert.Contains(t, event, "stopped")
-		assert.Contains(t, event, "zombie")
-		assert.Contains(t, event, "unknown")
-		total := event["sleeping"].(int) + event["running"].(int) + event["idle"].(int) +
-			event["stopped"].(int) + event["zombie"].(int) + event["unknown"].(int)
+	// if there's nothing marked as sleeping or idle, something weird is happening
+	assert.NotZero(t, event["total"])
 
-		assert.Equal(t, event["total"].(int), total)
+	var sum int
+	total := event["total"].(int)
+	for key, val := range event {
+		if key == "total" {
+			continue
+		}
+		sum = val.(int) + sum
 	}
+	assert.Equal(t, total, sum)
```

Comment:
Honestly, was largely done because I kinda felt bad about having _less_ tests. The fact that it no longer has a set list of states that it reports no matter what means I had to remove a good bit of testing logic, and I kinda felt like I needed to replace it with something.

Reply:
Right, two things you could check would be 1) that there aren't any unexpected keys/states in the map, 2) that there is at least one running process listed (this test at a minimum).